factcast-site/documentation-docsy/content/en/Usage/factus/projections/common-features.md

```diff
@@ -0,0 +1,65 @@
++++
+title = "Common Features"
+weight = 60
+type = "docs"
++++
+
+For a more fine-grained control of a projection, these optional hook methods can be implemented:
+
+| Method Signature                                 | Description            |
+|--------------------------------------------------|------------------------|
+| `List<FactSpec> postprocess(List<FactSpec> specsAsDiscovered)` | further filter the handled events via their [fact specification]({{<ref "factspec.md#specification">}}) including aggregate ID and meta entries | 
+| `void onCatchup()`                                | invoked after all past events of the event streams were processed. This is a good point to signal that the projection is ready to serve data (e.g. via a health indicator). |
+| `void onComplete()`                              | called when event processing completed. TODO how to differenciate between this and onCatchup?  |
```

Comment:
Thought about - if in practice `onComplete` is hardly used (since `onCatchup` is the more interesting hook) - maybe we just leave it out here.